common/persistence/visibility/elasticsearch/query/converter_test.go

```diff
@@ -0,0 +1,146 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Copyright (c) 2017 Xargin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package query is inspired and partially copied from by github.com/cch123/elasticsql.
+package query
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var errorCases = map[string]error{
+	"delete":                          MalformedSqlQueryErr,
+	"update x":                        MalformedSqlQueryErr,
+	"insert ":                         MalformedSqlQueryErr,
+	"insert into a values(1,2)":       NotSupportedErr,
+	"update a set id = 1":             NotSupportedErr,
+	"delete from a where id=1":        NotSupportedErr,
+	"select * from a where NOT(id=1)": NotSupportedErr,
+	"select * from a where 1 = 1":     InvalidExpressionErr,
+	"select * from a where 1=a":       InvalidExpressionErr,
+	"select * from a where zz(k=2)":   NotSupportedErr,
+	"select * from a group by k":      NotSupportedErr,
+	"select * from a where  a= 1 and multi_match(zz=1, query='this is a test', fields=(title,title.origin), type=phrase)": NotSupportedErr,
+}
+
+var supportedWhereCases = map[string]string{
+	"process_id= 1":                 `{"bool":{"filter":{"match_phrase":{"process_id":{"query":1}}}}}`,
+	"(process_id= 1)":               `{"bool":{"filter":{"match_phrase":{"process_id":{"query":1}}}}}`,
+	"((process_id= 1))":             `{"bool":{"filter":{"match_phrase":{"process_id":{"query":1}}}}}`,
+	"(process_id = 1 and status=1)": `{"bool":{"filter":[{"match_phrase":{"process_id":{"query":1}}},{"match_phrase":{"status":{"query":1}}}]}}`,
+	"`status`=1":                    `{"bool":{"filter":{"match_phrase":{"status":{"query":1}}}}}`,
```

Comment:
Why support quote with \`? Are `"'status'=1" or "\"status\"=1"` supported?

Reply:
`status` is a keyword in SQL.